docs/integrations/web.md

```diff
@@ -36,209 +36,177 @@ There are a couple important concepts to grasp for using Armory. We’ll look at
+        .add(new Input("name", "Type your name here...", "text"))
+        .group("bottom")
+        .add(new SubmitButton("submitName", "Done"))
+```
 
-How to store data? Well, the data the user inputs during the screen sequence are transferred as values of GET parameters with the corresponding `componentId` as key.
-We can store the user's answers at the end of the procedure. For example, here we only asked for the name, which the user set through an input area. 
-We can fetch it off the `Submit()` using `buttonId == "nameSubmitted"` because the `"nameSubmitted"` is the ID of the submit button that we will use as a trigger to take us to the next screen.
+We'll show you how to create these templates and custom components in the next section.
+
+Keep in mind that you can link together sequences of multiple Armory screens by specifying the transitions between them: the easiest way to do this is by setting the name of the next screen you want to go to as the value of the `SubmitButton` (e.g. `new SubmitButton("submitName", "Done", "askAddress")` takes the user to the screen where they will be asked to set their address). You can find plenty of examples of screen linking in the Armory [tutorial](/docs/tutorials/Armory-tutorial).
+
+## Creating components and templates
+
+### Custom components
+
+Let's take a look at how you can add new custom components to your screens. Let's say you wanted to add a component for uploading a document to your screen.
+The steps are as follows:
+1. Create the components directory in `services/armory/src/components` [TODO include by default?]
+2. Add the component in vue, for example create the file:
+```vue title="services/armory/src/components/FileUpload.vue"
+<template>
+  <div class="file is-boxed">
+    <label class="file-label">
+      <input class="file-input" type="file" name="resume" />
+      <span class="file-cta">
+        <span class="file-icon">
+          <i class="fas fa-upload"></i>
+        </span>
+        <span class="file-label"> Choose a file… </span>
+      </span>
+    </label>
+  </div>
+</template>
+
+<script>
+import BaseValue from "armory-sdk/src/components/base/BaseValue";
+
+export default {
+  name: "FileUpload",
+  extends: BaseValue,
+  props: {
+    placeholder: {
+      type: String,
+      default: "",
+    },
+  },
+};
+</script>
+
+<style></style>
+```
 
+:::tip
+The HTML you see in the `<template>...</template>` node was taken from this [website](https://bulma.io/documentation/). There are plenty of such resources out there, if you're not too crafty with HTML yourself!
+:::
 
-```java title="rule_engine/src/main/resources/rules/mindy/Mindy.drl"
+3. Finally, add your custom component to the `App.vue` file:
+```java title="services/armory/src/App.vue"
+...
+import FileUpload from "./components/FileUpload.vue";
 ...
+    getCustomComponents() {
+      return {FileUpload};
```

Comment:
I would color this line green to indicate `getCustomComponents` exists by default